walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,138 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+  /*
+   * Assigns attendees from request to local HashMap to search for them in constant time.
+   */
+  private HashSet<String> requestAttendeesToSet(MeetingRequest request) {
+    HashSet<String> requestAttendees = new HashSet<String>();
+    for (String attendee : request.getAttendees()) {
+      requestAttendees.add(attendee);
+    }
+    return requestAttendees;
+  }
+
+  /*
+   * Adds a TimeRange depending on the scenario. Some additions use a recursive call to this same
+   * function because if a new TimeRange is formed from a merge or an existing TimeRange is replaced
+   * because of a containment issue, it needs to be checked against all the previously added
+   * TimeRanges for new overlap or containment issues.
+   */
+  private ArrayList<TimeRange> addTimeRange(ArrayList<TimeRange> occupiedTimeSlots,
+      TimeRange addedTimeSlot) {
+    for (TimeRange timeSlot : occupiedTimeSlots) {
+      if (timeSlot.contains(addedTimeSlot)) {
```

Comment:
I forgot to say this before, but I love that you're utilizing the TimeRange API, good work doing that!

Reply:
Thanks!